packages/snowpack/src/commands/build.ts

```diff
@@ -52,25 +49,203 @@ async function installOptimizedDependencies(
+            contents: code,
+            locOnDisk: this.filepath,
+          };
+          break;
+        }
+      }
+
+      this.output[outLoc] = code;
+      if (map) {
+        this.output[path.join(this.outDir, sourceMappingURL)] = map;
+      }
+    }
+  }
+
+  async resolveImports(importMap: ImportMap) {
+    for (const [outLoc, file] of Object.entries(this.allFilesToResolveImports)) {
+      const resolveImportSpecifier = createImportResolver({
+        fileLoc: file.locOnDisk!, // we’re confident these are reading from disk because we just read them
+        dependencyImportMap: importMap,
+        config: this.config,
+      });
+      const resolvedCode = await transformFileImports(file, (spec) => {
+        // Try to resolve the specifier to a known URL in the project
+        let resolvedImportUrl = resolveImportSpecifier(spec);
+        if (!resolvedImportUrl || url.parse(resolvedImportUrl).protocol) {
+          return spec;
+        }
+        const extName = path.extname(resolvedImportUrl);
+        const isProxyImport = extName && extName !== '.js';
+        if (isProxyImport) {
+          resolvedImportUrl = resolvedImportUrl + '.proxy.js';
+        }
+        const isAbsoluteUrlPath = path.posix.isAbsolute(resolvedImportUrl);
+        const resolvedImportPath = removeLeadingSlash(path.normalize(resolvedImportUrl));
+
+        // We treat ".proxy.js" files special: we need to make sure that they exist on disk
+        // in the final build, so we mark them to be written to disk at the next step.
+        if (isProxyImport) {
+          if (isAbsoluteUrlPath) {
+            this.allImportProxyFiles.push(path.resolve(this.buildDirectoryLoc, resolvedImportPath));
+          } else {
+            this.allImportProxyFiles.push(path.resolve(path.dirname(outLoc), resolvedImportPath));
+          }
+        }
+        // When dealing with an absolute import path, we need to honor the baseUrl
+        if (isAbsoluteUrlPath) {
+          resolvedImportUrl = path
+            .relative(
+              path.dirname(outLoc),
+              path.resolve(this.buildDirectoryLoc, resolvedImportPath),
+            )
+            .replace(/\\/g, '/'); // replace Windows backslashes at the end, after resolution
+        }
+        // Make sure that a relative URL always starts with "./"
+        if (!resolvedImportUrl.startsWith('.') && !resolvedImportUrl.startsWith('/')) {
+          resolvedImportUrl = './' + resolvedImportUrl;
+        }
+        return resolvedImportUrl;
+      });
+      this.output[outLoc] = resolvedCode;
+    }
+  }
+
+  async writeToDisk() {
+    mkdirp.sync(this.outDir);
+    for (const [outLoc, code] of Object.entries(this.output)) {
+      console.log('writeToDisk', outLoc);
+      await fs.writeFile(outLoc, code, getEncodingType(path.extname(outLoc)));
+    }
+  }
+
+  async buildProxy() {
+    for (const importProxyFileLoc of this.allImportProxyFiles) {
+      if (existsSync(importProxyFileLoc)) {
```

Comment:
this is especially hacky, and doesn't really work without `--clean`. Will need to be cleaned up

Reply:
What is the scenario where overwriting is bad (and should it be a comment here)?

Reply:
already gone :)